packages/global-styles/src/provider.js

```diff
@@ -0,0 +1,189 @@
+/**
+ * External dependencies
+ */
+import { camelCase, kebabCase } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useContext, createContext } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { useRenderedGlobalStyles } from './renderer';
+import { fromPx, toPx } from './utils';
+
+const GlobalStylesContext = createContext( {} );
+export const useGlobalStylesContext = () => useContext( GlobalStylesContext );
+
+export function GlobalStylesProvider( { children, baseStyles, userEntityId } ) {
+	// Trigger entity retrieval.
+	useSelect( ( select ) =>
+		select( 'core' ).getEntityRecord(
+			'postType',
+			'wp_global_styles',
+			userEntityId
+		)
+	);
+	const globalStyles = useGlobalStyles( baseStyles, userEntityId );
+
+	return (
+		<GlobalStylesContext.Provider value={ globalStyles }>
+			{ children }
+		</GlobalStylesContext.Provider>
+	);
+}
+
+function toCase( tree, transformCase ) {
+	if ( ! ( tree instanceof Object ) ) {
+		return tree;
+	}
+
+	const newTree = {};
+	for ( const key in tree ) {
+		if ( ! tree.hasOwnProperty( key ) ) continue;
+
+		if ( tree[ key ] instanceof Object ) {
+			newTree[ transformCase( key ) ] = toCase(
+				tree[ key ],
+				transformCase
+			);
+		} else {
+			newTree[ transformCase( key ) ] = tree[ key ];
+		}
+	}
+	return newTree;
+}
+
+// Make object reference immutable
+// so it stays the same in the function setters.
+let styles = {};
```

Comment:
Should we move this inside the hook? And use the hook useRef with a start value of {}?

Reply:
Also, the comment says "Make object reference immutable", but it's using `let`, not `const`.

Reply:
cc: @nosolosw should we do some update here?